### PR TITLE
enemy manager and spawning implemented

### PR DESCRIPTION
**Pull request by Emmett-Byrne**

### CODE REVIEW

File: ARGO/include/EnemyManager.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+{
+public:
+	EnemyManager(SDL_Renderer* t_renderer, float t_initialSpawnDelay, EventManager& t_eventManager, PhysicsSystem& t_physicsSystem, CollisionSystem& t_collisionSystem, HealthSystem& t_healthSystem, AiSystem& t_aiSystem, RenderSystem& t_renderSystem, LevelManager& t_levelManager);
+	void init();
+	void update(float t_dt);
+	void render(SDL_Renderer* t_renderer);
+	void spawnGroup(float t_dt);
+	void createEnemyAtTile(Entity* t_tile);
+	Entity(&getEnemies())[Utilities::ENEMY_POOL_SIZE];
+	void killAlll();
```
Comment (CathalRedmond) [inline ARGO/include/EnemyManager.h]: 'Alll'

File: ARGO/include/GameScreen.h [hunk 2]
```diff
@@ -36,15 +36,14 @@ class GameScreen
...
-	void removeDeadEnemies();
 
 	EventManager& m_eventManager;
 
 	static const int MAX_ENTITIES = 10000;
 
 	Controller m_controllers[Utilities::S_MAX_PLAYERS];
 	Entity m_players[Utilities::S_MAX_PLAYERS];
-	std::vector<Entity> m_entities;
+	//std::vector<Entity> m_entities;
```
Comment (CathalRedmond) [inline ARGO/include/GameScreen.h]: Do you need this?

File: ARGO/src/EnemyManager.cpp [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	if (m_spawnEnemyTimer <= 0)
+	{
+		m_spawnEnemyTimer = SPAWN_ENEMY_RATE;
+		Entity* spawnTile = nullptr;
+		int attempts = 0;
+		while (spawnTile == nullptr && attempts < 20)
+		{
+			int x = glm::linearRand(-5, 5);
+			int y = glm::linearRand(-5, 5);
+			std::cout << x << ", " << y << std::endl;
```
Comment (CathalRedmond) [inline ARGO/src/EnemyManager.cpp]: what is this couting stuff

File: ARGO/src/EnemyManager.cpp [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+void EnemyManager::spawnGroup(float t_dt)
+{
+	m_spawnEnemyTimer -= t_dt;
+	if (m_spawnEnemyTimer <= 0)
+	{
+		m_spawnEnemyTimer = SPAWN_ENEMY_RATE;
+		Entity* spawnTile = nullptr;
+		int attempts = 0;
```
Comment (CathalRedmond) [inline ARGO/src/EnemyManager.cpp]: what does attempts do

File: ARGO/src/Game.cpp [hunk 5]
```diff
@@ -16,7 +16,7 @@ Game::Game() :
 	m_splashScreen{ m_eventManager },
 	m_mainMenuScreen{ m_eventManager },
 	m_achievementsScreen{ m_eventManager, m_controllers[0], m_renderer },
-	m_currentScreen{ MenuStates::Splash }
+	m_currentScreen{ MenuStates::Game }
```
Comment (CathalRedmond) [inline ARGO/src/Game.cpp]: reset to splash

File: ARGO/src/TransformComponent.cpp [hunk 6]
```diff
@@ -12,7 +12,9 @@ TransformComponent::TransformComponent(float t_x, float t_y, float t_rotation) :
 {
 	m_pos.x = t_x;
 	m_pos.y = t_y;
-	m_rotation = t_rotation;
+	m_rotation = t_rotation;
```
Comment (CathalRedmond) [inline ARGO/src/TransformComponent.cpp:15]: will you through in the comment from physics system that explains the minus 1 for left and right rotation or make a utilities value for it like int INVALID_ROTATION = -1

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

File: ARGO/include/GameScreen.h (hunk 2 above)
Comment (Violet-Byrne) [reply ARGO/include/GameScreen.h]: I knew I'd miss one of the things I commented out

File: ARGO/src/EnemyManager.cpp (hunk 4 above)
Comment (Violet-Byrne) [reply ARGO/src/EnemyManager.cpp]: Okay you can very clearly tell it's counting the amount of attempts for the while loop. I did put a maxAttempts var in their which better explains the code

File: ARGO/src/TransformComponent.cpp (hunk 6 above)
Comment (Violet-Byrne) [reply ARGO/src/TransformComponent.cpp:15]: What I'm wondering though is why you have 2 variables for this, why are you not using 1 variable where negative is one direction and positive is the other and 0 is neither?

File: ARGO/include/AiSystem.h [hunk 7]
```diff
@@ -7,14 +7,15 @@
...
+#include "HealthComponent.h"
 #include "AiStates.h"
 #include "Utilities.h"
 #include "BehaviourTree.h"
 
 class AiSystem : public BaseSystem
 {
 public:
-	AiSystem(Entity(&t_players)[Utilities::S_MAX_PLAYERS], std::vector<Entity>& t_enemies, EventManager& t_eventManager);
+	AiSystem(Entity(&t_players)[Utilities::S_MAX_PLAYERS], Entity(&t_enemies)[Utilities::ENEMY_POOL_SIZE] , EventManager& t_eventManager);
```
Comment (PrzemekTomczyk) [inline ARGO/include/AiSystem.h:18]: have a comment here please as this is a bit of a cluster fudge

Comment (Violet-Byrne) [reply ARGO/include/AiSystem.h:18]: This is the proper syntax for passing in a reference to an array of a specific size. Does it need a comment?

**Pull request merged**